lib/src/code_generator/objc_interface.dart

```diff
@@ -247,8 +280,12 @@ class ObjCInterface extends BindingType {
       }
       s.write('    $selName ??= '
           '${_builtInFunctions.registerName}(_lib, "${m.originalName}");\n');
-      final convertReturn = _needsConverting(returnType);
-      s.write('    ${convertReturn ? 'final _ret = ' : 'return '}');
+      final convertReturn = m.kind != ObjCMethodKind.propertySetter &&
```

Comment:
Maybe this check should be whether the return type is void, rather than checking if its a setter?